### PR TITLE
New runtime & gamestate

### PR DESCRIPTION
**Pull request by gecotron**
Based on previous work & discussions. Please let me know any tweaks & changes needed

### CODE REVIEW

File: .github/workflows/rust-windows.yml [hunk 1]
```diff
@@ -2,7 +2,7 @@ name: Rust Windows
 
 on:
   push:
-    branches: ["main"]
```
Comment (phynxinferno) [inline .github/workflows/rust-windows.yml:5]: Maybe change this back on merge

**Comment by phynxinferno** [comment]
Looks good so far. Let's avoid merging for now, I'll test some changes at home.

File: src/main.rs [hunk 2]
```diff
@@ -94,3 +70,62 @@ fn window_conf() -> Conf {
...
+            function.call();
+        }
+    }
+
+    /// Push value as mutable reference
+    /// For player, maps & other gameObjects
+    fn push_mut<T: Call>(&mut self, item: &'s mut T) {
+        self.function_stack.push(Box::new(item));
+    }
+
```
Comment (phynxinferno) [inline src/main.rs:220]: Looking good so far but we need to make the drawstack calls more ambiguous

File: .github/workflows/rust-windows.yml (hunk 1 above)
Comment (gecotron) [reply .github/workflows/rust-windows.yml:5]: :+1:

File: src/main.rs (hunk 2 above)
Comment (gecotron) [reply src/main.rs:220]: Will do. Function pointer passing is a thing now with more recent commits on the branch, if we can, I think updating/making a new pull request will probably be in order soon. Not immediately though, there is still some work to be done

**Comment by gecotron** [comment]
Quick note. This probably should not be merged, as the mdw-gamestate branch has more updates to the system, and a myriad of other work there that should be all dumped in one go. Another pr will probably be made

**Comment by gecotron** [comment]
Closed due to language switch

**Pull request closed without merging**